tools/kind-d8.sh

```diff
@@ -0,0 +1,502 @@
+      sleep 3
+    fi
+  done
+}
+configs_create() {
+  echo "Creating ${CONFIG_DIR} directory..."
+  mkdir -p ${CONFIG_DIR}
+
+  echo "Creating kind config file (${CONFIG_DIR}/kind.cfg)..."
+  cat <<EOF >${CONFIG_DIR}/kind.cfg
+apiVersion: kind.x-k8s.io/v1alpha4
+kind: Cluster
+nodes:
+- role: control-plane
+  extraPortMappings:
+  - containerPort: 80
+    hostPort: 80
+    listenAddress: "127.0.0.1"
+    protocol: TCP
+  - containerPort: 443
+    hostPort: 443
+    listenAddress: "127.0.0.1"
+    protocol: TCP
+EOF
+
+  echo "Creating Deckhouse installation config file (${CONFIG_DIR}/config.yml)..."
+  cat <<EOF >${CONFIG_DIR}/config.yml
+apiVersion: deckhouse.io/v1
+kind: InitConfiguration
+deckhouse:
+  releaseChannel: Stable
+  bundle: Minimal
+  configOverrides:
+    global:
+      modules:
+        publicDomainTemplate: "%s-127-0-0-1.nip.io"
+        https:
+          mode: Disabled
+    operatorPrometheusCrdEnabled: true
+    operatorPrometheusEnabled: true
+    prometheusCrdEnabled: true
+    prometheusEnabled: true
+    prometheus:
+      longtermRetentionDays: 0
+    monitoringKubernetesControlPlaneEnabled: true
+    ingressNginxEnabled: true
+EOF
+
+  if [[ -n "$D8_LICENSE_KEY" ]]; then
+    generate_ee_access_string "$D8_LICENSE_KEY"
+    cat <<EOF >>${CONFIG_DIR}/config.yml
+  imagesRepo: $D8_REGISTRY_PATH
+  registryDockerCfg: $D8_EE_ACCESS_STRING
+EOF
+  fi
+
+  echo "Creating Deckhouse resource file (${CONFIG_DIR}/resources.yml)..."
+  cat <<EOF >${CONFIG_DIR}/resources.yml
+apiVersion: deckhouse.io/v1
+kind: IngressNginxController
+metadata:
+  name: nginx
+spec:
+  ingressClass: nginx
+  inlet: HostPort
+  hostPort:
+    httpPort: 80
+    httpsPort: 443
+EOF
+}
+
+cluster_create() {
+
+  ${KIND_PATH} create cluster --name "${KIND_CLUSTER_NAME}" --image "${KIND_IMAGE}" --config "${CONFIG_DIR}/kind.cfg"
+
+  if [ "$?" -ne "0" ]; then
+    printf "
+Error creating cluster. If error is like '...port is already allocated' then you need to free ports 80 and 443.
+E.g., you can find programs that use these ports using the following command:
+
+    sudo lsof -n -i TCP@0.0.0.0:80,443 -s TCP:LISTEN
+
+"
+    exit 1
+  fi
+
+  ${KIND_PATH} get kubeconfig --internal --name "${KIND_CLUSTER_NAME}" >${CONFIG_DIR}/kubeconfig
+
+}
+
+deckhouse_install() {
+  echo "Running Deckhouse installation..."
+
+  docker run --pull=always --rm --network kind -v "${CONFIG_DIR}/config.yml:/config.yml" -v "${CONFIG_DIR}/resources.yml:/resources.yml" \
+    -v "${CONFIG_DIR}/kubeconfig:/kubeconfig" ${D8_REGISTRY_PATH}/install:$D8_UPDATE_CHANNEL \
+    bash -c "dhctl bootstrap-phase install-deckhouse --kubeconfig=/kubeconfig --kubeconfig-context=kind-${KIND_CLUSTER_NAME} --config=/config.yml && \
+             dhctl bootstrap-phase create-resources --kubeconfig=/kubeconfig --kubeconfig-context=kind-${KIND_CLUSTER_NAME} --resources=/resources.yml"
+
+  if [ "$?" -ne "0" ]; then
+    echo "Error installing Deckhouse!"
+    exit 1
+  fi
+}
+
+ingress_check() {
```

Comment:
You can made it a `bootstrap-script` to remove kubectl dependency. Could you please discuss it with @name212 ?

Reply:
Never mind, this is not possible. I thought that we could use the master node kubectl, but it looks like there is no option (or options are too complicated).